packages/core/lib/commands/init/initSource/truffle-config.js

```diff
@@ -113,8 +113,12 @@ module.exports = {
       //    enabled: false,
       //    runs: 200
       //  },
-      //  evmVersion: "byzantium"
-      // }
+      //  evmVersion: "byzantium",
+      // },
+      // disableDefaults: false, // Disable the default solc optimizer settings; if true,
+      //                         // whatever setting is passed in settings.optimizer will be respected;
+      //                         // if false, the default when settings.optimizer is missing is
```

Comment:
This doesn't _only_ apply to the optimizer settings, right? We're disabling defaults for all solc settings?

Reply:
The issue was specifically the optimizer. I actually was torn on this and looked at my notes from our call with Harry. Specifically optimizer being empty was being replaced with { enabled: false } and this was a problem pre-Solidity 0.8.6. I think it would be a mistake to disable all default settings if this flag is set. Like, I'm not sure what the evmVersion stuff is about and didn't think we should mess it up. 

Perhaps I should rename this to `disableOptimizerDefaults` to be more clear?

Reply:
Why don't we just make it about all default solc settings? Seems silly to scope this to just that one field.

Reply:
(I know that right now, the optimizer defaults are what's causing problems, but I could see other defaults being problematic in the future)

Reply:
If we want to scope this more narrowly, then we should definitely rename it to reflect this. This is not my vote, but if others are cool with having a separate `disableOptimizerDefaults` (understanding that we might also need a broader flag in the future)

Reply:
The optimizer default is a bug and this fix is meant to address it in a non-breaking way. In general I would not expect us to provide defaults when they are not necessary to the proper running of our code and thus chose this narrow approach.

For instance, having `remappings: []` as a default likely makes a difference. It certainly affects behavior, I believe. If you set `remappings` to an empty array, you are explicitly telling the compiler to use only the default remappings and disallowing the use of any custom mappings. If you leave out the remappings setting entirely, you are still using the default remappings, but leaving open the possibility of using custom mappings in the future. I think it is tricky!

Reply:
I'm just thinking, like, if we need to do another one of these in the future, it's not like the `if` statements we're leaving in place here for `disableOptimizerDefaults` will ever go away. At best, we're looking at a field we'll want to deprecate, at worst, a permanent logical complexity for future contributors.

vs. `disableDefaults`, which just turns off all the defaults Truffle provides... a simple notion!

Reply:
Sure, but disabling all default settings could be a breaking change in some circumstances. Well, _would_ be for someone in the scenario I wrote out above.

Reply:
Yeah, but it's inherently "breaking" if you tell Truffle you don't want the values it's been giving you. All it means is that we'll need to add `remappings: []` ourselves inside the Blockscout fetcher.

Reply:
We also should not need to do this again. It is a hack, and I'm not sure we should code toward needing similar hacks in the future... We should not add defaults that we then need to disable, I think.

Reply:
I see your argument though that `disableDefaults` introduces this caveat-emptor can of worms. But I don't think it's a big deal because people looking to disable defaults will almost certainly have their fully-populated settings.

Reply:
Well, this isn't just about that though. The main reason we are doing this is for _already deployed_ contracts we are fetching from Blockscout. We don't know what the preferred configuration was in that setting and we could be introducing a new bug when trying to address the optimizer bug.

Reply:
Making the setting user-facing was a secondary decision to the main purpose of this code change. In fact, I'd probably not make it user-facing, since defaults only happen when you don't set it yourself. Not sure why someone would want optimizer to be `{}` instead of `enabled: false` and it just sort of pollutes the config. 🤷‍♀️

Reply:
There's been this long outstanding issue with Truffle that we don't let users just wholesale tell us what solc settings to use. This optimizer thing is only the latest of these, so that's why I thought we were talking about the problem generally, since that's a similar diff and solves a categorical concern. Solving this problem one at a time just adds technical debt in the form of `if` statements and the need to maintain these nuanced configuration params.

Anyway, hopefully my argument on the matter is clear. I'll abstain further to let others comment if they have thoughts.

Reply:
I'd be happy to hear other perspectives of course, but just wanted to address the previous Truffle issues linked here because I do think they actually support my points! 😄 

[#2355](https://github.com/trufflesuite/truffle/issues/2355) actually led to [this PR](https://github.com/trufflesuite/truffle/pull/2358) to _add_ the default `remappings` setting to `configDefaults` and pass it along the flow for Truffle's compilation. But wait! [This issue](https://github.com/trufflesuite/truffle/issues/2768) was created the following year, pointing out that Truffle's resolver is possibly _unable_ to resolve remappings passed through the settings! 🤯 Thus, _the default setting of `remappings: []` may be necessary_. AFAIK, nobody ever fixed the remappings issue with the resolver. And I don't think that (priority 4) issue should block this PR, which is attempting to fix a very specific defaults bug. 🤷‍♀️ 

As for the other two issues shared, the cause was never completely sorted out for the first, and the third is what this PR is attempting to address. 🎉 

I hope that is some helpful history on this rather more complicated than expected default settings issue.

Reply:
Oh also, happy with the option of just disabling the defaults altogether, but that would be a breaking change and we need this fix short-term so we can enable Blockscout support. 🤷‍♀️